lib/internal/Magento/Framework/Encryption/Adapter/Mcrypt.php

```diff
@@ -0,0 +1,164 @@
+                    new \Magento\Framework\Phrase('Key must not exceed %1 bytes.', [$maxKeySize])
+                );
+            }
+            // @codingStandardsIgnoreStart
+            $initVectorSize = @mcrypt_enc_get_iv_size($this->handle);
+            // @codingStandardsIgnoreEnd
+            if (true === $initVector) {
+                /* Generate a random vector from human-readable characters */
+                $abc = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+                $initVector = '';
+                for ($i = 0; $i < $initVectorSize; $i++) {
+                    $initVector .= $abc[rand(0, strlen($abc) - 1)];
+                }
+            } elseif (false === $initVector) {
+                /* Set vector to zero bytes to not use it */
+                $initVector = str_repeat("\0", $initVectorSize);
+            } elseif (!is_string($initVector) || strlen($initVector) != $initVectorSize) {
+                throw new \Magento\Framework\Exception\LocalizedException(
+                    new \Magento\Framework\Phrase(
+                        'Init vector must be a string of %1 bytes.',
+                        [$initVectorSize]
+                    )
+                );
+            }
+            $this->_initVector = $initVector;
+        } catch (\Exception $e) {
+            // @codingStandardsIgnoreStart
+            @mcrypt_module_close($this->handle);
+            // @codingStandardsIgnoreEnd
+            throw $e;
+        }
+        // @codingStandardsIgnoreStart
+        @mcrypt_generic_init($this->handle, $key, $initVector);
+        // @codingStandardsIgnoreEnd
+    }
+
+    /**
+     * Destructor frees allocated resources
+     */
+    public function __destruct()
+    {
+        // @codingStandardsIgnoreStart
+        @mcrypt_generic_deinit($this->handle);
+        // @codingStandardsIgnoreEnd
+        // @codingStandardsIgnoreStart
+        @mcrypt_module_close($this->handle);
+        // @codingStandardsIgnoreEnd
+    }
+
+    /**
+     * Retrieve a name of currently used cryptographic algorithm
+     *
+     * @return string
+     */
+    public function getCipher()
+    {
+        return $this->cipher;
+    }
+
+    /**
+     * Mode in which cryptographic algorithm is running
+     *
+     * @return string
+     */
+    public function getMode()
+    {
+        return $this->mode;
+    }
+
+    /**
+     * Retrieve an actual value of initial vector that has been used to initialize a cipher
+     *
+     * @return string
+     */
+    public function getInitVector()
+    {
+        return $this->initVector;
+    }
+
+    /**
+     * Encrypt a data
+     *
+     * @param  string $data String to encrypt
+     * @return string
+     * @throws \Exception
+     */
+    public function encrypt($data)
+    {
+        throw new \Exception(
+            (string)new \Magento\Framework\Phrase('Mcrypt cannot be used for encryption. Use Sodium instead')
```

Comment:
If this is logic exception, no need to wrap the message into `Phrase` object and add it to the dictionary.